openpype/hosts/houdini/api/pipeline.py

```diff
@@ -35,71 +40,138 @@
-    register_loader_plugin_path(LOAD_PATH)
-    register_creator_plugin_path(CREATE_PATH)
-
-    log.info("Installing callbacks ... ")
-    # register_event_callback("init", on_init)
-    register_event_callback("before.save", before_save)
-    register_event_callback("save", on_save)
-    register_event_callback("open", on_open)
-    register_event_callback("new", on_new)
-
-    pyblish.api.register_callback(
-        "instanceToggled", on_pyblish_instance_toggled
-    )
-
-    self._has_been_setup = True
-    # add houdini vendor packages
-    hou_pythonpath = os.path.join(HOUDINI_HOST_DIR, "vendor")
-
-    sys.path.append(hou_pythonpath)
-
-    # Set asset settings for the empty scene directly after launch of Houdini
-    # so it initializes into the correct scene FPS, Frame Range, etc.
-    # todo: make sure this doesn't trigger when opening with last workfile
-    _set_context_settings()
-    shelves.generate_shelves()
-
-
-def uninstall():
-    """Uninstall Houdini-specific functionality of avalon-core.
-
-    This function is called automatically on calling `api.uninstall()`.
-    """
+class HoudiniHost(HostBase, IWorkfileHost, ILoadHost, INewPublisher):
+    name = "houdini"
+
+    def __init__(self):
+        super(HoudiniHost, self).__init__()
+        self._op_events = {}
+        self._has_been_setup = False
+
+    def install(self):
+        pyblish.api.register_host("houdini")
+        pyblish.api.register_host("hython")
+        pyblish.api.register_host("hpython")
+
+        pyblish.api.register_plugin_path(PUBLISH_PATH)
+        register_loader_plugin_path(LOAD_PATH)
+        register_creator_plugin_path(CREATE_PATH)
+
+        log.info("Installing callbacks ... ")
+        # register_event_callback("init", on_init)
+        self._register_callbacks()
+        register_event_callback("before.save", before_save)
+        register_event_callback("save", on_save)
+        register_event_callback("open", on_open)
+        register_event_callback("new", on_new)
+
+        pyblish.api.register_callback(
+            "instanceToggled", on_pyblish_instance_toggled
+        )
+
+        self._has_been_setup = True
+        # add houdini vendor packages
+        hou_pythonpath = os.path.join(HOUDINI_HOST_DIR, "vendor")
+
+        sys.path.append(hou_pythonpath)
+
+        # Set asset settings for the empty scene directly after launch of
+        # Houdini so it initializes into the correct scene FPS,
+        # Frame Range, etc.
+        # TODO: make sure this doesn't trigger when
+        #       opening with last workfile.
+        _set_context_settings()
+        shelves.generate_shelves()
+
+    def has_unsaved_changes(self):
+        return hou.hipFile.hasUnsavedChanges()
+
+    def get_workfile_extensions(self):
+        return [".hip", ".hiplc", ".hipnc"]
+
+    def save_workfile(self, dst_path=None):
+        # Force forwards slashes to avoid segfault
+        filepath = dst_path.replace("\\", "/")
```

Comment:
`dst_path=None` default wouldn't work because you can't perform `.replace` on None type.

Reply:
Also feels weird that the argument is `dst_path` but `open_workfile` uses `filepath` as argument.

Should this maybe be `dest_filepath` or just `filepath`?